✨ endpoints

fagustin07 opened the pull request.

fagustin07 committed:
✨ search reviews endpoint with filters

nicolasmartinez0510 commented on src/main/kotlin/ar/edu/unq/grupoN/backenddesappapi/service/ReviewService.kt:
```diff
@@ -43,12 +54,113 @@ class ReviewService{
+
+
+        //filtering platform
+        if (applicableFilters.platform != null) {
+            val platform = cb.equal(review.get<String>("platform"), applicableFilters.platform)
+            predicates.add(platform)
+        }
```
Creo que hay muchos de estos filtrados que repiten un patron bastante parecido. Se podria hacer una subtarea donde le pases lo que queres filtrar, el string del campo, el builder, la review y la lista. O algo mas feo, usar una estructura de control tipo for. Incluso se podria refactorizar en mas de una subtarea. 
Pero es un refactor boludo que no aporta funcionalidad, es mas de "reducir lineas de codigo".

nicolasmartinez0510 commented on src/main/kotlin/ar/edu/unq/grupoN/backenddesappapi/service/ReviewService.kt:
```diff
@@ -43,12 +54,113 @@ class ReviewService{
+
+
+        //order by date
+        if (applicableFilters.orderByDate) {
+            if (applicableFilters.order == "DESC") {
+                orders.add(cb.desc(review.get<LocalDateTime>("date")))
+            } else {
+                orders.add(cb.asc(review.get<LocalDateTime>("date")))
+            }
+        }
+
+        //order by valoration
+        if (applicableFilters.orderByRating) {
+            if (applicableFilters.order == "DESC") {
+                orders.add(cb.desc(review.get<Int>("valoration")))
+            } else {
+                orders.add(cb.asc(review.get<Int>("valoration")))
+            }
+        }
```
Este es un caso parecido al del comentario anterior, se podria refactorizar en una subtarea para no repetir tanto codigo

nicolasmartinez0510 commented on src/main/kotlin/ar/edu/unq/grupoN/backenddesappapi/model/exceptions.kt:
```diff
@@ -2,10 +2,12 @@ package ar.edu.unq.grupoN.backenddesappapi.model
 
-class DoesNotExistChapterException(message: String): ReviewTypeException(message)
+class DoesNotExistChapterException(title: String) :
+    ReviewTypeException("This chapter doesn't exist in $title yet")
```
Hay un test que esta fallando porque faltaron las ' ' en $title. Se podria o bien sacarlas del test o bien agregarlas al texto de la excepcion. Las dos valen aunque prefiero quedarme con las ' '

fagustin07 replied:
lo corrijo para el proximo push

nicolasmartinez0510 commented:
En lineas generales me parecio muy copado el uso de Criteria, haciendo algun que otro refactor para reutilizar codigo repetido esta para mergear  👍

fagustin07 committed:
✨ report review model, tests, service and endpoint
smalls refactors

fagustin07 committed:
✨ hidden a review with 5 reports or more + add new report types

fagustin07 committed:
basic platform admin client register and login + refactors

fagustin07 committed:
refactors + base reverse search + moving querybuilder in a repository

fagustin07 committed:
finished reverse search endpoint + refactor in model to filter by reviews ratings

fagustin07 committed:
refactors:
more legible methods in repository custom.
rename file Filters
refactors on Swagger descriptions
delete unused enum

fagustin07 committed:
delete unused code and refactor in filter cast member

fagustin07 committed:
aspect to handle endpoints exceptions
